src/github-url.js

```diff
@@ -0,0 +1,131 @@
+const parseGithubUrl = require('parse-github-url');
+const fetch = require('node-fetch');
+
+const githubAPI = 'http://api.github.com';
+
+
+const isValidGithubUrl = (ghUrl) => ghUrl.protocol === 'https:'
+                                  && ghUrl.host === 'github.com'
+                                  && (ghUrl.branch !== 'issues'
+                                    || ghUrl.branch !== 'pull'
+                                    || ghUrl.branch !== 'master');
+
+/**
+ * @description Takes a url, checks if it's a valid github url
+ * and returns JSON with fetched data related to the url from Github's V3 API.
+ * At the moment, it accepts urls of repositories, issues andpull requests.
+ * @link https://developer.github.com/v3/
```

Comment:
Question: why not use GitHub's JS wrapper for this API, https://octokit.github.io/rest.js/?  It has a lot of convenience things already done for you.  We could file a follow-up issue later to refactor to use it.

Reply:
I wasn't aware of it at all! I'll file an issue to refactor this using octokit/rest.js once it gets merged.